flask/helpers.py

```diff
@@ -553,7 +538,9 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         rv.cache_control.max_age = cache_timeout
         rv.expires = int(time() + cache_timeout)
 
-    if add_etags and filename is not None:
+    if add_etags and filename is not None and file is None:
```

Comment:
Here, `file` can only be `None` if
- filename was provided
- current_app.use_x_sendfile is `True` (see line 513)

I don't think that was the point of this patch, unless there's something I'm missing